tests/unit/authenticators/oauth2-password-grant-test.js

```diff
@@ -127,6 +127,34 @@ describe('OAuth2PasswordGrantAuthenticator', () => {
       authenticator.authenticate('username', 'password');
     });
 
+    it('sends an AJAX request to the token endpoint without client_id Basic Auth header when enableLegacyClientIdentification is false', function(done) {
+      server.post('/token', (request) => {
+        expect(request.requestHeaders.hasOwnProperty('authorization')).to.eql(false);
+        done();
+      });
+
+      authenticator.set('clientId', 'test-client');
+      authenticator.set('enableLegacyClientIdentification', false);
+      authenticator.authenticate('username', 'password');
+    });
+
+    it('sends an AJAX request to the token endpoint with client_id in POST body when enableLegacyClientIdentification is false', function(done) {
+      server.post('/token', (request) => {
+        let body = parsePostData(request.requestBody);
+        expect(body).to.eql({
+          'grant_type': 'password',
+          'username': 'username',
+          'password': 'password',
+          'client_id': 'test-client'
+        });
+        done();
+      });
+
+      authenticator.set('clientId', 'test-client');
+      authenticator.set('enableLegacyClientIdentification', false);
+      authenticator.authenticate('username', 'password');
+    });
```

Comment:
I think we could combine the two tests since they are essentially doing the same thing but asserting on different parts of the request

Reply:
I'd prefer leaving assertions for separate aspects in separate tests.